back/app/services/llm_selector.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+# Service to select the most appropriate and permitted LLM for a given task
+class LLMSelector
+  LLMFamily = Data.define(:key, :name, :description)
+  LLMUseCase = Data.define(:key, :description, :supported_models, :default_model)
+
+  FAMILIES = [
+    LLMFamily.new(
+      key: 'vertex_gemini',
+      name: 'Google Gemini (Vertex AI)',
+      description: 'Google Gemini models via Vertex AI'
+    ),
+    LLMFamily.new(
+      key: 'azure_openai',
+      name: 'Azure OpenAI',
+      description: 'OpenAI models via Microsoft Azure OpenAI Service'
+    ),
+    LLMFamily.new(
+      key: 'aws_anthropic',
+      name: 'AWS Bedrock Anthropic Claude',
+      description: 'Anthropic Claude models via AWS Bedrock'
+    )
+  ]
+
+  MODELS = [
+    ::Analysis::LLM::GPT4oMini,
+    ::Analysis::LLM::GPT4o,
+    ::Analysis::LLM::GPT41,
+    ::Analysis::LLM::ClaudeHaiku45,
+    ::Analysis::LLM::ClaudeSonnet45,
+    ::Analysis::LLM::ClaudeOpus45,
+    ::Analysis::LLM::Gemini25Pro,
+    ::Analysis::LLM::Gemini25Flash
+  ]
+
+  USE_CASES = [
+    LLMUseCase.new(
+      key: 'sensemaking_auto_tagging',
+      description: 'Automatic tagging of content in sensemaking',
+      supported_models: [::Analysis::LLM::GPT4oMini, ::Analysis::LLM::ClaudeHaiku45, ::Analysis::LLM::Gemini25Flash],
+      default_model: ::Analysis::LLM::GPT4oMini
+    ),
+    LLMUseCase.new(
+      key: 'sensemaking_summarization',
+      description: 'Summarization of content in sensemaking',
+      supported_models: [::Analysis::LLM::GPT41, ::Analysis::LLM::Gemini25Pro, ::Analysis::LLM::ClaudeOpus45],
+      default_model: ::Analysis::LLM::GPT41
+    ),
+    LLMUseCase.new(
+      key: 'toxicity_detection',
+      description: 'Detection of toxic content',
+      supported_models: [::Analysis::LLM::ClaudeHaiku45, ::Analysis::LLM::GPT4oMini, ::Analysis::LLM::Gemini25Flash],
+      default_model: ::Analysis::LLM::ClaudeHaiku45
+    ),
+    LLMUseCase.new(
+      key: '360_input_file_description',
+      description: 'Generating descriptions for 360 input files',
+      supported_models: [::Analysis::LLM::GPT4o, ::Analysis::LLM::ClaudeSonnet45, ::Analysis::LLM::Gemini25Pro],
+      default_model: ::Analysis::LLM::GPT4o
+    ),
+    LLMUseCase.new(
+      key: 'idea_feed_live_topic_model',
+      description: 'Automatically manage the tags in the Idea Feed constantly.',
+      supported_models: [::Analysis::LLM::ClaudeOpus45, ::Analysis::LLM::Gemini25Pro],
+      default_model: ::Analysis::LLM::Gemini25Pro
+    ),
+    LLMUseCase.new(
+      key: 'idea_feed_live_classification',
+      description: 'Classify ideas into existing topics in the Idea Feed constantly.',
+      supported_models: [::Analysis::LLM::ClaudeHaiku45, ::Analysis::LLM::Gemini25Flash],
+      default_model: ::Analysis::LLM::Gemini25Flash
+    )
+  ]
+
+  def llm_claz_for_use_case(use_case_key, app_configuration = AppConfiguration.instance)
```

Comment:
(nitpick: I only use `klass` or similar when it’s the full variable name.)
```suggestion
  def llm_class_for_use_case(use_case_key, app_configuration = AppConfiguration.instance)
```